frontend/src/AppBuilder/_stores/slices/eventsSlice.js

```diff
@@ -1062,6 +1096,33 @@ export const createEventsSlice = (set, get) => ({
         return executeAction(event, mode, {});
       };
 
+      const logInfo = (log) => {
+        const error = new Error();
+        const stackLine = error.stack.split('\n')[2];
+        const lineNumberMatch = stackLine.match(/:(\d+):\d+\)$/);
+        const lineNumber = lineNumberMatch ? lineNumberMatch[1] : 'unknown';
+        const event = { actionId: 'log-info', description: `${log}, Line ${lineNumber - 2}`, eventType: 'customLog' };
+        return executeAction(event, mode, {});
+      };
+
+      const logError = (log) => {
+        const error = new Error();
+        const stackLine = error.stack.split('\n')[2];
+        const lineNumberMatch = stackLine.match(/:(\d+):\d+\)$/);
+        const lineNumber = lineNumberMatch ? lineNumberMatch[1] : 'unknown';
+        const event = { actionId: 'log-error', description: `${log}, Line ${lineNumber - 2}`, eventType: 'customLog' };
+        return executeAction(event, mode, {});
+      };
+
+      const log = (log) => {
+        const error = new Error();
+        const stackLine = error.stack.split('\n')[2];
+        const lineNumberMatch = stackLine.match(/:(\d+):\d+\)$/);
+        const lineNumber = lineNumberMatch ? lineNumberMatch[1] : 'unknown';
+        const event = { actionId: 'log', description: `${log}, Line ${lineNumber - 2}`, eventType: 'customLog' };
+        return executeAction(event, mode, {});
+      };
+
```

Comment:
Here as well. We can create a common function and use it

Reply:
These functions need to be defined separately since i need to use the stackLine to identify where the log is being picked from, If I define a common function then the stackLine displayed would be something like 1338 which would be the actual file line number instead of the runjs code linenumber